pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -407,6 +406,7 @@ def training_type_plugin(self) -> TrainingTypePlugin:
         # attach checkpoint plugin to the training type plugin
         if self._checkpoint_io is not None:
             self._training_type_plugin.checkpoint_io = self._checkpoint_io
+        self._training_type_plugin._precision_plugin = self.precision_plugin
```

Comment:
Perhaps we need to None check here too?

Reply:
If it's none, in training_type_plugin init() it will be set to Precision_plugin(), should we raise misconfig() or runtimeError if precision_plugin is None though? I feel just default Precision_plugin() could cause silent failures?

Reply:
Actually it can never be None. In the AcceleratorConnector.select_precision_plugin we always return a plugin and have a misconfiguration when no precision is detected. So we should be fine here.